src/Workspaces/Core/Portable/Workspace/Host/CompilationFactory/ICompilationFactoryService.cs

```diff
@@ -10,12 +12,9 @@ namespace Microsoft.CodeAnalysis.Host
     internal interface ICompilationFactoryService : ILanguageService
     {
         Compilation CreateCompilation(string assemblyName, CompilationOptions options);
-        Compilation CreateSubmissionCompilation(string assemblyName, CompilationOptions options, Type hostObjectType);
-        Compilation GetCompilationFromCompilationReference(MetadataReference reference);
```

Comment:
Why are these removed? Is it related to this PR?

Reply:
Turns out these were unused; it was easier to remove them than null annotate them.

Reply:
i'm a fan of pulling those sorts of chnages out into a simultaneous PR that is easy to review/merge.  jsut a thought for the future.

Reply:
This was broken down commit-by-commit; my bad for not mentioning that in the original PR message.